app/routes/endemics/date-of-visit-ms.js

```diff
@@ -211,6 +218,40 @@ const postHandler = {
 
       const dateOfVisit = new Date(request.payload[labels.year], request.payload[labels.month] - 1, request.payload[labels.day])
 
+      if (isDairy && isEndemicsFollowUp && isBeforeDairyFollowUpRelease(dateOfVisit)) {
+        raiseInvalidDataEvent(
+          request,
+          dateOfVisitKey,
+          `User is attempting to claim for dairy follow-up with a date of visit of ${dateOfVisit} which is before dairy follow-ups was enabled.`
+        )
+
+        setEndemicsClaim(request, dateOfVisitKey, dateOfVisit)
+
+        return h
+          .view(endemicsDairyFollowUpDateException, { backLink: pageUrl, ruralPaymentsAgency: config.ruralPaymentsAgency })
+          .code(400)
+          .takeover()
+      }
+
+      if (previousClaims.length > 0) {
+        const speciesChanged = Boolean(previousClaims.find(claim => claim.data.typeOfLivestock !== typeOfLivestock))
```

Comment:
maybe use `some` over `find` then no need to wrap in Boolean.

Reply:
good call